### ISSUE TITLE
for vs for...of vs forEach()

### ISSUE THREAD

**Issue by lorenzo-lipparini** [lorenzo-lipparini/CNR-project#2]
Currently, in both **menger-sponge** and **koch-snowflake**, the methods which iterate over the child elements of fractals use a `for...of` loop, preferring it to alternatives like the Array `forEach()` method or even a plain `for` loop.

*But is this really the best choice?*

Array methods like `forEach()` may improve code readability, while on the other hand an "uglier" `for` loop might give a significant performance boost.

Consider this simple algorithm, which takes the square root of all the numbers in a list and assigns them to a variable:
```javascript
let list = [ /* a lot of numbers */ ];
let test = 0;

for (let i = 0; i < list.length; i++) {
  test = Math.sqrt(list[i]);
}
```

As it turns out, the time it takes for this algorithm to finish changes a lot depending on the way you iterate over the list:

|    `for`    | `for...of` | `forEach()` |
|:---------:|:---------:|:-----------:|
| 2326ms | 3427ms |  3185ms   |

*(tested in Chrome, list containing 50'000'000 items)*

In conclusion, the `for` loop implementation outspeeds the other two, with the `for...of` being even slower than the `forEach()`, and this gain in performance might actually be worth the loss in readability.

**Comment by lorenzo-lipparini** [comment]
All the `for...of` loops in the project were replaced with a `for` loop  in [this branch](https://github.com/lorenzo-lipparini/CNR-project/tree/for-loop).

This new version of the project was then tested for performance and compared to the original one in two different scenarios:

- Rendering the **increasing-iterations** animation of **menger-sponge** up to the third iteration
- Rendering a single image of a 5th-iteration *Menger Sponge*

In both cases, the time required turned out to be about the same for both implementations, varying by few seconds (which is a negligeable difference, given that that the actual rendering was minutes long).

In the end, sticking with the `for...of` loop seems the best thing to do for now.
